addon/model.js

```diff
@@ -75,6 +75,13 @@ class EmbeddedInternalModel {
   }
 }
 
+function _computeAttributeReference(key, value, modelName, schemaInterface, schema) {
+  schemaInterface._keyBeingResolved = key;
```

Comment:
I'm generally a fan of hiding internal state manipulation behind properly named functions:
```js
schemaInterface._beginKeyResolution(key);
let reference = schema.computeAttributeReference(key, value, modelName, schemaInterface);
schemaInterface._endKeyResolution(key);
```
Later, when you need to expand the properties to notify:
```js
let dependentKeys = schemaInterface._getDependentKeys(key);
if (dependentKeys) {
  this._notifyProperties(dependentKeys);
}
```

This makes it explicitly what is the intention of `_keyBeingResolved` property is and the added complexity to implement the functionality is kept to a minimum in places of the code, which are already quite complex.

Reply:
Thank you @dnachev. it's a good point and looks clean. I will update based on above suggestions.